### PR TITLE
Yuwei/bugfix 202211

### PR DESCRIPTION
**Pull request by ivywei0125**
What does this implement/fix? Explain your changes.
---------------------------------------------------
bugfix

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java [hunk 1]
```diff
@@ -226,8 +226,8 @@ public void handle(Object source, Object e) {
                     AzureTaskManager.getInstance().runLater(() -> openSettingsAction.handle(null, e));
                 }
             };
-            final String INSTALL_SUCCEED_MESSAGE = "Download and install Azure Functions Core Tools(%s) successfully. Auto configured Azure Functions Core Tools path in Azure Settings";
-            AzureMessager.getMessager().success(String.format(INSTALL_SUCCEED_MESSAGE, azureEvent.getSource().toString()), "Install succeed", openSettingsActionInMessage);
+            final String INSTALL_SUCCEED_MESSAGE = "Download and install Azure Functions Core Tools successfully. Auto configured Azure Functions Core Tools path in Azure Settings";
+            AzureMessager.getMessager().success(INSTALL_SUCCEED_MESSAGE, "Install successfully", openSettingsActionInMessage);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java]: remove the title

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/resources/bundles/com/microsoft/azure/toolkit/operation.properties [hunk 2]
```diff
@@ -62,7 +62,7 @@ function.clean_staging_folder.folder=clean up staging folder({0})
 function.complete_deployment.app=complete the deployment of function({0}) and refresh Azure Explorer
 function.complete_run.func=complete running function({0})
 function.copy_settings.settings|folder=copy local settings({0}) to staging folder({1})
-function.get_version.func=get version of function({0})
+function.get_version.func=get version of Azure Functions Core Tools({0})
```
Comment (wangmingliang-ms) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/resources/bundles/com/microsoft/azure/toolkit/operation.properties:65]: null check the parameter before using this operation.

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java [hunk 3]
```diff
@@ -79,45 +80,47 @@ private void doDownloadReleaseTools(@Nullable ReleaseInfo releaseInfo, String do
...
+            if (zipEntry.isDirectory()) {
+                createIfNotExist(zipEntryFile.getAbsolutePath());
+            } else {
+                final FileOutputStream fileOutputStream = new FileOutputStream(zipEntryFile);
+                final byte[] buffer = new byte[1024 * 10];
+                int len = 0;
+                while ((len = zipInputStream.read(buffer)) != -1) {
+                    fileOutputStream.write(buffer, 0, len);
+                }
+                fileOutputStream.close();
```
Comment (wangmingliang-ms) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java:117]: use `try with resource`

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java [hunk 4]
```diff
@@ -79,45 +80,47 @@ private void doDownloadReleaseTools(@Nullable ReleaseInfo releaseInfo, String do
...
-                len = zipInputStream.read(buffer);
+            if (zipEntry.isDirectory()) {
+                createIfNotExist(zipEntryFile.getAbsolutePath());
+            } else {
+                final FileOutputStream fileOutputStream = new FileOutputStream(zipEntryFile);
+                final byte[] buffer = new byte[1024 * 10];
+                int len = 0;
+                while ((len = zipInputStream.read(buffer)) != -1) {
+                    fileOutputStream.write(buffer, 0, len);
+                }
```
Comment (wangmingliang-ms) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java:114]: use `try with resource` and refer to 
https://stackoverflow.com/questions/30335332/apache-commons-unzip-method

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationService.java [hunk 5]
```diff
@@ -59,7 +59,7 @@ public ExperimentationService create() {
     private void updateFeatures() {
         try {
             final Response response = client.newCall(request).execute();
```
Comment (Flanker32) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationService.java]: May use try with resource here to ensure response is closed, refers https://github.com/square/okhttp/blob/master/okhttp/src/jvmMain/kotlin/okhttp3/Call.kt#L51

**Review by wangmingliang-ms — APPROVED** [review]
LGTM

**Pull request merged**